### PR TITLE
Add support for service accounts

### LINKED ISSUE #1266: Support for `datadog_service_account` and `datadog_service_account_application_key` resource (and data source)
I recently knew that Datadog has it's own service account and can create a API key. Before this, the API keys were attached to a person who created the key so in my org, some retired member still owned a key. 

This feature makes easy to managed the keys in orgs.

### Affected Resource(s)

None. This request adds new source and data source.

## Referencens

* [Create an application key for this service account](https://docs.datadoghq.com/api/latest/service-accounts/)

### PR DESCRIPTION
**Pull request by jake-scott**
Resolves #1266

### CODE REVIEW

File: docs/resources/application_key.md [hunk 1]
```diff
@@ -26,6 +26,10 @@ resource "datadog_application_key" "foo" {
 
 - **name** (String) Name for Application Key.
 
+### Optional
+
+- **service_account** (String) ID of a service account that will own the Application Key
```
Comment (alai97) [inline docs/resources/application_key.md:31]:
```suggestion
- **service_account** (String) ID of a service account that owns the Application Key.
```

**Review by alai97 — DISMISSED** [review]
Looks good pending one copy nit!

File: docs/resources/application_key.md (hunk 1 above)
Comment (jake-scott) [reply docs/resources/application_key.md:31]: Thanks, fixed that..

Comment (alai97) [reply docs/resources/application_key.md:31]: @jake-scott I don't think you added this suggestion in, try one more time.

Comment (jake-scott) [reply docs/resources/application_key.md]: ah sorry, forgot the 'make docs', should be good now..

**Review by alai97 — APPROVED** [review]
Thanks for opening this PR @jake-scott, looks good for docs!

**Comment by srosenthal-dd** [comment]
Hi! I'm part of the team (AAA) that owns users/service accounts at Datadog. I'm hoping have someone on my team (maybe me) look at this within the next week or so. Thanks for your patience.

**Comment by jake-scott** [comment]
> Hi! I'm part of the team (AAA) that owns users/service accounts at Datadog. I'm hoping have someone on my team (maybe me) look at this within the next week or so. Thanks for your patience.

No hurry from my perspective.. thanks.

**Comment by jake-scott** [comment]
Hi - any update on this item?

**Comment by srosenthal-dd** [comment]
Sorry, I totally blew through my expected one week deadline, eh? Sorry! It's still on my todo list. Hoping to take a closer look soon. My team is getting up to speed on the Terraform provider now.

**Comment by swetha-murali-dd** [comment]
I am not sure about this but I am wondering if we need to add support for service account app keys in data_source_datadog_application_keys.go. I am leaning towards no because for service account app keys, only key name is returned and not the key value and I don't think key name can be meaningfully used anywhere. 
@srosenthal-dd let me know what you think.

**Comment by NasAmin** [comment]
Any news on this please?  we'd love to have this supported to use the new Observability Pipelines across multiple environments.

**Comment by skarimo** [comment]
Superseded by https://github.com/DataDog/terraform-provider-datadog/pull/1490

**Pull request closed without merging**